### PR TITLE
added new sensors to mes23xx

### PR DESCRIPTION
**Pull request by Npeca75**
1. PoE temperature
2. PoE values
3. SFP modules inventory (model/serial)
4. stacked chassis count
5. grouped sensors
6. automatic discovery of SFP dbm/volt/bias Alarm values
7. skip backup battery detection if not connected
8. skip SFP detection in case of Copper SFP-s, zero vaules
9. chassis PSU status
10. chassis temperature status

devices covered
            - .1.3.6.1.4.1.35265.1.112 #MES 2308R
            - .1.3.6.1.4.1.35265.1.107 #MES 2348P
            - .1.3.6.1.4.1.35265.1.96 #MES 3348
            - .1.3.6.1.4.1.35265.1.83 #MES 2324B
            - .1.3.6.1.4.1.35265.1.81 #MES 3324F
            - .1.3.6.1.4.1.35265.1.80 #MES 3324
            - .1.3.6.1.4.1.35265.1.78 #MES 2324FB AC
            - .1.3.6.1.4.1.35265.1.74 #MES 5324

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [x] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [x] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.
- [x] If my Pull Request makes discovery/polling/yaml changes, I have added/updated [test data](https://docs.librenms.org/Developing/os/Test-Units/).

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Comment by Npeca75** [comment]
lnms1
lnms2
lnms3
lnms4
lnms5
lnms6

File: includes/discovery/entity-physical/eltex-mes23xx.inc.php [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+            'entPhysicalContainedIn'  => $connectedto,
+            'entPhysicalMfgName'      => $data['eltPhdTransceiverInfoVendorName'],
+            'entPhysicalHardwareRev'  => normData($data['eltPhdTransceiverInfoVendorRev']),
+            'entPhysicalIsFRU'        => 'true',
+        ];
+    }
+}
+
+foreach ($entity_array as $entPhysicalIndex => $entry) {
+    $entPhysicalIndex = array_key_exists('entPhysicalIndex', $entry) ? $entry['entPhysicalIndex'] : '';
```
Comment (Jellyfrog) [inline includes/discovery/entity-physical/eltex-mes23xx.inc.php:64]: for next time, you can simplify it like:
```suggestion
    $entPhysicalIndex = $entry['entPhysicalIndex'] ?? '';
```

Comment (Jellyfrog) [reply includes/discovery/entity-physical/eltex-mes23xx.inc.php:64]: also, if these fields can be null they should be rather than a empty  string

**Comment by Jellyfrog** [comment]
Seems to miss a couple of json test data files

File: includes/discovery/entity-physical/eltex-mes23xx.inc.php (hunk 1 above)
Comment (Npeca75) [reply includes/discovery/entity-physical/eltex-mes23xx.inc.php:64]: @Jellyfrog 
last time i spoke with @murrant he said there is no need for json, but no problem, uploded one

Comment (Npeca75) [reply includes/discovery/entity-physical/eltex-mes23xx.inc.php]: @Jellyfrog 
done

**Comment by Jellyfrog** [comment]
We don't need/want snmprecs without corresponding jsons. what murrant means is that we dont need test data for very similar devices

**Comment by Npeca75** [comment]
> We don't need/want snmprecs without corresponding jsons. what murrant means is that we dont need test data for very similar devices

ok @Jellyfrog 
i am not a native English speaker, so i might misunderstood what @murrant said about JSONs.

uploaded JSONs for each snmpsim
hope this is OK

**Comment by murrant** [comment]
> We don't need/want snmprecs without corresponding jsons. what murrant means is that we dont need test data for very similar devices

@Jellyfrog yes, but if we are adding additional device sysObjectIDs and want to test detection only, we do want to add just an snmprec with sysObjectID and sysDescr.   This will test detection but not add a lot of time to overall tests testing basically the same thing.

**Comment by Npeca75** [comment]
@murrant  @Jellyfrog 
done,
removed JSONs, leave only essential for sensors discovery
fixed conflicted YAML

File: includes/definitions/eltex-mes23xx.yaml [hunk 2]
```diff
@@ -22,3 +27,8 @@ bad_iftype:
     - tunnel
     - softwareLoopback
     - propPointToPointSerial
+
+oids:
+    no_bulk:
+        - rlPethPsePortOutputPower
```
Comment (murrant) [inline includes/definitions/eltex-mes23xx.yaml:33]: Can you please change these to use the fully qualified OID name?
`MARVELL-POE-MIB::rlPethPsePortOutputPower`
`MARVELL-POE-MIB::rlPethPsePortPowerLimit`

(and use that in the snmpwalk too)

Comment (Npeca75) [reply includes/definitions/eltex-mes23xx.yaml:33]: > Can you please change these to use the fully qualified OID name? `MARVELL-POE-MIB::rlPethPsePortOutputPower` `MARVELL-POE-MIB::rlPethPsePortPowerLimit`
> 
> (and use that in the snmpwalk too)

function gen_snmpwalk_cmd($device, $oids, $options = null, $mib = null, $mibdir = null)
{
    d_echo(Config::getCombined($device['os'], 'oids.no_bulk'));

output:    
 rlPethPsePortOutputPower 
array (
  0 => 'MARVELL-POE-MIB::rlPethPsePortOutputPower',
  1 => 'MARVELL-POE-MIB::rlPethPsePortPowerLimit',
)  

so array is there

but looks like it is matched only aganist $oids
|| ! empty(array_intersect(Arr::wrap($oids), Config::getCombined($device['os'], 'oids.no_bulk'))))) 

maybe i am wrong ...

Comment (Npeca75) [reply includes/definitions/eltex-mes23xx.yaml]: > Can you please change these to use the fully qualified OID name? `MARVELL-POE-MIB::rlPethPsePortOutputPower` `MARVELL-POE-MIB::rlPethPsePortPowerLimit`
> 
> (and use that in the snmpwalk too)

ok, yes, you are right
if i put MARVELL-POE-MIB:: in walk, then MARVELL-POE-MIB:: will work in YAML
sorry for noise

**Pull request merged**